source/operations/getDavCompliance.ts

```diff
@@ -0,0 +1,39 @@
+import { joinURL } from "../tools/url.js";
+import { encodePath } from "../tools/path.js";
+import { request, prepareRequestOptions } from "../request.js";
+import { handleResponseCode } from "../response.js";
+import {
+    WebDAVMethodOptions,
+    WebDAVClientContext,
+    WebDAVClientError,
+    DavCompliance
+} from "../types.js";
+
+export async function getDavCompliance(
+    context: WebDAVClientContext,
+    filePath: string,
+    options: WebDAVMethodOptions = {}
+): Promise<DavCompliance> {
+    const requestOptions = prepareRequestOptions(
+        {
+            url: joinURL(context.remoteURL, encodePath(filePath)),
+            method: "OPTIONS"
+        },
+        context,
+        options
+    );
+    const response = await request(requestOptions);
+    try {
+        handleResponseCode(context, response);
+    } catch (err) {
+        const error = err as WebDAVClientError;
+        throw error;
```

Comment:
Why catch if we're just throwing again? Maybe wrap in `Layerr` to give more insight/context? Improving the error message with Layerr would be fine.

Reply:
It seems that there is not much contextual information here.

I've just removed `try ... catch ...`